winml/test/model/model_tests.cpp

```diff
@@ -0,0 +1,212 @@
+          actualOutput = Ort::Value::CreateTensor(
+              memoryInfo,
+              actualData,
+              actualSizeInBytes,
+              expectedShapeAndTensorType.GetShape().data(),
+              expectedShapeAndTensorType.GetShape().size(),
+              expectedShapeAndTensorType.GetElementType()));
+
+      // Use the expected and actual OrtValues to compare
+      std::pair<COMPARE_RESULT, std::string> ret = CompareOrtValue(*actualOutput, *value, m_perSampleTolerance, m_relativePerSampleTolerance, m_postProcessing);
+      WINML_EXPECT_EQUAL(COMPARE_RESULT::SUCCESS, ret.first) << ret.second;
+    }
+  }
+};
+
+TEST_P(ModelTest, Run) {
+  LearningModel model = nullptr;
+  LearningModelDevice device = nullptr;
+  LearningModelSession session = nullptr;
+  LearningModelBinding binding = nullptr;
+  WINML_EXPECT_NO_THROW(model = LearningModel::LoadFromFilePath(m_testCase->GetModelUrl()));
+  WINML_EXPECT_NO_THROW(device = LearningModelDevice(m_deviceKind));
+  WINML_EXPECT_NO_THROW(session = LearningModelSession(model, device));
+  WINML_EXPECT_NO_THROW(binding = LearningModelBinding(session));
+  for (size_t i = 0; i < m_testCase->GetDataCount(); i++) {
+    // Load and bind inputs
+    onnxruntime::test::HeapBuffer inputHolder;
+    std::unordered_map<std::string, Ort::Value> inputFeeds;
+    WINML_EXPECT_NO_THROW(m_testCase->LoadTestData(i, inputHolder, inputFeeds, true));
+    WINML_EXPECT_NO_THROW(BindInputsFromFeed(binding, inputFeeds));
+
+    // evaluate
+    LearningModelEvaluationResult results = nullptr;
+    WINML_EXPECT_NO_THROW(results = session.Evaluate(binding, L"Testing"));
+
+    // Load expected outputs
+    onnxruntime::test::HeapBuffer outputHolder;
+    std::unordered_map<std::string, Ort::Value> outputFeeds;
+    WINML_EXPECT_NO_THROW(m_testCase->LoadTestData(i, outputHolder, outputFeeds, false));
+
+    // compare results
+    CompareEvaluationResults(results, outputFeeds);
+  }
+}
+
+// Get the path of the model test collateral. Will return empty string if it doesn't exist.
+std::string GetTestDataPath() {
+  std::string testDataPath(MAX_PATH, '\0');
+  auto environmentVariableFetchSuceeded = GetEnvironmentVariableA("WINML_TEST_DATA_PATH", testDataPath.data(), MAX_PATH);
+  if (environmentVariableFetchSuceeded == 0 && GetLastError() == ERROR_ENVVAR_NOT_FOUND || environmentVariableFetchSuceeded > MAX_PATH) {
+    // if the WINML_TEST_DATA_PATH environment variable cannot be found, attempt to find the hardcoded models folder
+    std::wstring modulePath = FileHelpers::GetModulePath();
+    std::filesystem::path currPath = modulePath.substr(0,modulePath.find_last_of(L"\\"));
+    std::filesystem::path parentPath = currPath.parent_path();
+    auto hardcodedModelPath = parentPath.string() + "\\models";
+    if (std::filesystem::exists(hardcodedModelPath) && hardcodedModelPath.length() <= MAX_PATH) {
+      return hardcodedModelPath;
+    }
+  }
+  return std::move(testDataPath);
```

Comment:
That's a pessimizing move (see https://stackoverflow.com/questions/62061433/how-to-avoid-the-pessimizing-move-warning-of-nrvo). clang and gcc warn about it, msvc doesn't yet.